File: cpp/program12.cpp
// progra57.c

#include<iostream>
using namespace std;

int CheckDigit(int iNo)
{
    int iDigit = 0;
    int iCount = 0;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        if(iDigit == 7)
        {
            iCount++;
        }
        iNo = iNo / 10;
    }
    return iCount;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number"<<"\n";
    cin>>iValue;

    iRet = CheckDigit(iValue);
    cout<<"Frequency of 7 is "<<iRet<<"\n" ;

    return 0;
}

File: scala/program12.scala
import scala.io.StdIn.{readInt}

object Demo 
{
    def checkDigit(num : Int): Int = {
        var count = 0
        var temp = num;
        while(temp != 0)
        {
            var digit = temp % 10
            if(digit == 7) count += 1
            temp /= 10
        }
        return count
    }

    def main(args : Array[String])   
    {
        println("Enter number : ")
        var num : Int = readInt()

        var freq = checkDigit(num);
        println("Frequency of the 7 is : "+freq)
    }
}